<h3>Data set #4:<br>
Economic entities in operation at the beginning of the year (Total by employees) <br>
http://osp.stat.gov.lt/en/statistiniu-rodikliu-analize?portletFormName=visualization&hash=62026300-429b-41af-9712-3070f6ad8868</h3>

In [50]:
#pulling csv data from file
import csv
filename='Source_datasets/Exported matrix table_d4.csv'
with open(filename, 'r') as f:
    data = list(csv.reader(f))
#print (data)
#print (f)
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', 'Economic entities in operation at the beginning of the year | units']
['Total by employees', 'Republic of Lithuania', '2016', '99200']
['', 'Alytus county', '2016', '3503']
['', 'Alytus t. mun.', '2016', '1749']
['', 'Alytus d. mun.', '2016', '432']
['', 'Druskininkai mun.', '2016', '642']
['', 'Lazdijai d. mun.', '2016', '284']
['', 'Varėna d. mun.', '2016', '396']
['', 'Kaunas county', '2016', '19857']
['', 'Birštonas mun.', '2016', '107']
['', 'Jonava d. mun.', '2016', '803']
['', 'Kaišiadorys d. mun.', '2016', '575']
['', 'Kaunas c. mun.', '2016', '13676']
['', 'Kaunas d. mun.', '2016', '2508']
['', 'Kėdainiai d. mun.', '2016', '1086']
['', 'Prienai d. mun.', '2016', '470']
['', 'Raseiniai d. mun.', '2016', '632']
['', 'Klaipėda county', '2016', '11141']
['', 'Klaipėda c. mun.', '2016', '6508']
['', 'Klaipėda d. mun.', '2016', '1576']
['', 'Kretinga d. mun.', '2016', '912']
['', 'Neringa mun.', '2016', '226']
['', 'Palanga t. mun.', '2016', '703']
['', 'Skuodas 

In [51]:
#remove aggregated rows, as aggregation will take place later e.g. in pivotjs
keysList=['Republic','county']
tobedeletedList=[]
rangeCurser={}#to tag _ layers 
updated_lengh=len(data)
for row_num in range(0,updated_lengh):
    for key in keysList:
        #print (key)
        if key in data[row_num][1]:
            tobedeletedList.append(row_num)
            if 'Republic' in key:
                rangeCurser[data[row_num][0]]=row_num
                
import operator

sorted_rangeCurser = sorted(rangeCurser.items(), key=operator.itemgetter(1))

print (tobedeletedList)
print (rangeCurser)
print('sorted cursor:{}'.format(sorted_rangeCurser))


for aggregated_row_num in tobedeletedList:
    #print (data[aggregated_row_num])
    data[aggregated_row_num]=['']
    #tobedeletedList[:] = [x - 1 for x in tobedeletedList]
   #print (tobedeletedList)
    
for row_num in range(0,len(data)):
    print (data[row_num])


[1, 2, 8, 17, 25, 32, 39, 47, 52, 57, 64, 73, 74, 80, 89, 97, 104, 111, 119, 124, 129, 136, 145, 146, 152, 161, 169, 176, 183, 191, 196, 201, 208, 217, 218, 224, 233, 241, 248, 255, 263, 268, 273, 280, 289, 290, 296, 305, 313, 320, 327, 335, 340, 345, 352, 361, 362, 368, 377, 385, 392, 399, 407, 412, 417, 424, 433, 434, 440, 449, 457, 464, 471, 479, 484, 489, 496, 505, 506, 512, 521, 529, 536, 543, 551, 556, 561, 568, 577, 578, 584, 593, 601, 608, 615, 623, 628, 633, 640, 649, 650, 656, 665, 673, 680, 687, 695, 700, 705, 712, 721, 722, 728, 737, 745, 752, 759, 767, 772, 777, 784]
{'Total by employees': 1, '1 000 and more employees': 721, '100–149 employees': 433, '50–99 employees': 361, '150–249 employees': 505, '10–19 employees': 217, '250–499 employees': 577, '20–49 employees': 289, '500–999 employees': 649, '0–4 employees': 73, '5–9 employees': 145}
sorted cursor:[('Total by employees', 1), ('0–4 employees', 73), ('5–9 employees', 145), ('10–19 employees', 217), ('20–49 employees', 

In [52]:
#remove first layer as it is aggregated (total by employee)
totalbyEmployeeLayer_start_end=[sorted_rangeCurser[0][1],sorted_rangeCurser[1][1]]
print (totalbyEmployeeLayer_start_end)

del_curser=totalbyEmployeeLayer_start_end[1]-1 #-1 is important not to delete the start of the next layer
print (del_curser)

for i in range(totalbyEmployeeLayer_start_end[0],totalbyEmployeeLayer_start_end[1]):
    print ("delete row {}:{}".format(del_curser,data[del_curser]))
    data[del_curser]=['']
    del_curser=del_curser-1
    
for row_num in range(0,len(data)):
    print (data[row_num])


[1, 73]
72
delete row 72:['', 'Vilnius d. mun.', '2016', '2383']
delete row 71:['', 'Vilnius c. mun.', '2016', '35275']
delete row 70:['', 'Ukmergė d. mun.', '2016', '690']
delete row 69:['', 'Trakai d. mun.', '2016', '700']
delete row 68:['', 'Švenčionys d. mun.', '2016', '333']
delete row 67:['', 'Širvintos d. mun.', '2016', '272']
delete row 66:['', 'Šalčininkai d. mun.', '2016', '479']
delete row 65:['', 'Elektrėnai mun.', '2016', '417']
delete row 64:['']
delete row 63:['', 'Zarasai d. mun.', '2016', '272']
delete row 62:['', 'Visaginas mun.', '2016', '457']
delete row 61:['', 'Utena d. mun.', '2016', '923']
delete row 60:['', 'Molėtai d. mun.', '2016', '405']
delete row 59:['', 'Ignalina d. mun.', '2016', '236']
delete row 58:['', 'Anykščiai d. mun.', '2016', '468']
delete row 57:['']
delete row 56:['', 'Telšiai d. mun.', '2016', '915']
delete row 55:['', 'Rietavas mun.', '2016', '175']
delete row 54:['', 'Plungė d. mun.', '2016', '800']
delete row 53:['', 'Mažeikiai d. mun.', '2

<p>sorted cursor:[('Total by employees', 1), ('0–4 employees', 73), ('5–9 employees', 145), ('10–19 employees', 217), ('20–49 employees', 289), ('50–99 employees', 361), ('100–149 employees', 433), ('150–249 employees', 505), ('250–499 employees', 577), ('500–999 employees', 649), ('1 000 and more employees', 721)]</p>

In [53]:
#add missing ranges layers values
first_Layer_curser=[sorted_rangeCurser[1][1],sorted_rangeCurser[2][1]]
first_Layer_tag=sorted_rangeCurser[1][0]

second_Layer_curser=[sorted_rangeCurser[2][1],sorted_rangeCurser[3][1]]
second_Layer_tag=sorted_rangeCurser[2][0]

third_Layer_curser=[sorted_rangeCurser[3][1],sorted_rangeCurser[4][1]]
third_Layer_tag=sorted_rangeCurser[3][0]

fourth_Layer_curser=[sorted_rangeCurser[4][1],sorted_rangeCurser[5][1]]
fourth_Layer_tag=sorted_rangeCurser[4][0]

fifth_Layer_curser=[sorted_rangeCurser[5][1],sorted_rangeCurser[6][1]]
fifth_Layer_tag=sorted_rangeCurser[5][0]

sixth_Layer_curser=[sorted_rangeCurser[6][1],sorted_rangeCurser[7][1]]
sixth_Layer_tag=sorted_rangeCurser[6][0]

seventh_Layer_curser=[sorted_rangeCurser[7][1],sorted_rangeCurser[8][1]]
seventh_Layer_tag=sorted_rangeCurser[7][0]

eight_Layer_curser=[sorted_rangeCurser[8][1],sorted_rangeCurser[9][1]]
eight_Layer_tag=sorted_rangeCurser[8][0]

nineth_Layer_curser=[sorted_rangeCurser[9][1],sorted_rangeCurser[10][1]]
nineth_Layer_tag=sorted_rangeCurser[9][0]

tenth_Layer_curser=[sorted_rangeCurser[10][1],len(data)]
tenth_Layer_tag=sorted_rangeCurser[10][0]


print(first_Layer_curser)
print(second_Layer_curser)
print(nineth_Layer_curser)
print(tenth_Layer_tag)

for i in range(first_Layer_curser[0],first_Layer_curser[1]):
    data[i][0]=first_Layer_tag

for i in range(second_Layer_curser[0],second_Layer_curser[1]):
    data[i][0]=second_Layer_tag

for i in range(third_Layer_curser[0],third_Layer_curser[1]):
    data[i][0]=third_Layer_tag
    
for i in range(fourth_Layer_curser[0],fourth_Layer_curser[1]):
    data[i][0]=fourth_Layer_tag

for i in range(fifth_Layer_curser[0],fifth_Layer_curser[1]):
    data[i][0]=fifth_Layer_tag

for i in range(sixth_Layer_curser[0],sixth_Layer_curser[1]):
    data[i][0]=sixth_Layer_tag

for i in range(seventh_Layer_curser[0],seventh_Layer_curser[1]):
    data[i][0]=seventh_Layer_tag

for i in range(eight_Layer_curser[0],eight_Layer_curser[1]):
    data[i][0]=eight_Layer_tag

for i in range(nineth_Layer_curser[0],nineth_Layer_curser[1]):
    data[i][0]=nineth_Layer_tag

for i in range(tenth_Layer_curser[0],tenth_Layer_curser[1]):
    data[i][0]=tenth_Layer_tag
    
    
for row_num in range(0,len(data)):
    print (data[row_num])

[73, 145]
[145, 217]
[649, 721]
1 000 and more employees
['\ufeff', '', '', 'Economic entities in operation at the beginning of the year | units']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['']
['0–4 employees']
['0–4 employees']
['0–4 employees', 'Alytus t. mun.', '2016', '1141']
['0–4 employees', 'Alytus d. mun.', '2016', '283']
['0–4 employees', 'Druskininkai mun.', '2016', '430']
['0–4 employees', 'Lazdijai d. mun.', '2016', '175']
['0–4 employees', 'Varėna d. mun.', '2016', '222']
['0–4 employees']
['0–4 employees', 'Birštonas mun.', '2016', '60']
['0–4 employees', 'Jonava d. mun.', '2016', '447']
['0–4 employees', 'Kaišiadorys d. mun.', '2016', '331']
['0–4 employees', 'Ka

In [54]:
#permanent clean of unwanted rows in dataset
tobedeletedList=[]
for row_num in range (1,len(data)):
    if (len(data[row_num])<3):#regarding temporary header size
        #print ("row num{}:{}".format(i,data[row_num]))
        tobedeletedList.append(row_num)
print (tobedeletedList)
for aggregated_row_num in tobedeletedList:
    #print (data[aggregated_row_num])
    del data[aggregated_row_num]
    tobedeletedList[:] = [x - 1 for x in tobedeletedList]
   #print (tobedeletedList)
    
for row_num in range(0,len(data)):
    print (data[row_num])

[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 80, 89, 97, 104, 111, 119, 124, 129, 136, 145, 146, 152, 161, 169, 176, 183, 191, 196, 201, 208, 217, 218, 224, 233, 241, 248, 255, 263, 268, 273, 280, 289, 290, 296, 305, 313, 320, 327, 335, 340, 345, 352, 361, 362, 368, 377, 385, 392, 399, 407, 412, 417, 424, 433, 434, 440, 449, 457, 464, 471, 479, 484, 489, 496, 505, 506, 512, 521, 529, 536, 543, 551, 556, 561, 568, 577, 578, 584, 593, 601, 608, 615, 623, 628, 633, 640, 649, 650, 656, 665, 673, 680, 687, 695, 700, 705, 712, 721, 722, 728, 737, 745, 752, 759, 767, 772, 777, 784]
['\ufeff', '', '', 'Economic entities in operation at the beginning of the year | units']
['0–4 employees', 'Alytus t. mun.', '2016', '1141']
['0–4 employees', 'Alytus d. mun.

In [55]:
#add missing country values
for row_num in range(1,len(data)):
    data[row_num].append('Republic of Lithuania')
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', 'Economic entities in operation at the beginning of the year | units']
['0–4 employees', 'Alytus t. mun.', '2016', '1141', 'Republic of Lithuania']
['0–4 employees', 'Alytus d. mun.', '2016', '283', 'Republic of Lithuania']
['0–4 employees', 'Druskininkai mun.', '2016', '430', 'Republic of Lithuania']
['0–4 employees', 'Lazdijai d. mun.', '2016', '175', 'Republic of Lithuania']
['0–4 employees', 'Varėna d. mun.', '2016', '222', 'Republic of Lithuania']
['0–4 employees', 'Birštonas mun.', '2016', '60', 'Republic of Lithuania']
['0–4 employees', 'Jonava d. mun.', '2016', '447', 'Republic of Lithuania']
['0–4 employees', 'Kaišiadorys d. mun.', '2016', '331', 'Republic of Lithuania']
['0–4 employees', 'Kaunas c. mun.', '2016', '8433', 'Republic of Lithuania']
['0–4 employees', 'Kaunas d. mun.', '2016', '1545', 'Republic of Lithuania']
['0–4 employees', 'Kėdainiai d. mun.', '2016', '718', 'Republic of Lithuania']
['0–4 employees', 'Prienai d. mun.', '2016', '262', 'Republ

In [56]:
#add missing county values
chuncks=[0,61,122,183,244,305,366,427,488,549]

for i in range(0,10):
    for row_num in range(1+chuncks[i],6+chuncks[i]):
            data[row_num].append('Alytus county')
    for row_num in range(6+chuncks[i],14+chuncks[i]):
            data[row_num].append('Kaunas county')
    for row_num in range(14+chuncks[i],21+chuncks[i]):
            data[row_num].append('Klaipeda county')
    for row_num in range(21+chuncks[i],27+chuncks[i]):
            data[row_num].append('Marijampole county')
    for row_num in range(27+chuncks[i],33+chuncks[i]):
            data[row_num].append('Panevezys county')
    for row_num in range(33+chuncks[i],40+chuncks[i]):
            data[row_num].append('Siauliai county')
    for row_num in range(40+chuncks[i],44+chuncks[i]):
            data[row_num].append('Taurage county')
    for row_num in range(44+chuncks[i],48+chuncks[i]):
            data[row_num].append('Telsiai county')
    for row_num in range(48+chuncks[i],54+chuncks[i]):
            data[row_num].append('Utena county')
    for row_num in range(54+chuncks[i],62+chuncks[i]):
            data[row_num].append('Vilnius county')

        
for row_num in range(0,len(data)):
    print (data[row_num])

['\ufeff', '', '', 'Economic entities in operation at the beginning of the year | units']
['0–4 employees', 'Alytus t. mun.', '2016', '1141', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Alytus d. mun.', '2016', '283', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Druskininkai mun.', '2016', '430', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Lazdijai d. mun.', '2016', '175', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Varėna d. mun.', '2016', '222', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Birštonas mun.', '2016', '60', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Jonava d. mun.', '2016', '447', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Kaišiadorys d. mun.', '2016', '331', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Kaunas c. mun.', '2016', '8433', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Kaunas d. mun.', '2016', '1545', 'Republic 

In [57]:
#Adding collumn headers istead of the melformed header
headerList=['employee_number','Municipality','Time_period','Economic_entities_in_operation_at_the_beginning_of_the_year_units','Country','County']
del data[0]
data.insert( 0, headerList)
for row_num in range(0,len(data)):
    print (data[row_num])

['employee_number', 'Municipality', 'Time_period', 'Economic_entities_in_operation_at_the_beginning_of_the_year_units', 'Country', 'County']
['0–4 employees', 'Alytus t. mun.', '2016', '1141', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Alytus d. mun.', '2016', '283', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Druskininkai mun.', '2016', '430', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Lazdijai d. mun.', '2016', '175', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Varėna d. mun.', '2016', '222', 'Republic of Lithuania', 'Alytus county']
['0–4 employees', 'Birštonas mun.', '2016', '60', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Jonava d. mun.', '2016', '447', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Kaišiadorys d. mun.', '2016', '331', 'Republic of Lithuania', 'Kanuas county']
['0–4 employees', 'Kaunas c. mun.', '2016', '8433', 'Republic of Lithuania', 'Kanuas county']
['0–4 emplo

In [58]:
import csv
#print (data)
with open("Ready_datasets/readyDataSet4.csv", "w") as f:
    writer = csv.writer(f)
    writer.writerows(data)

<h4>construct  query</h4>
<p>EEOBY_Query=PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>\n\<br>
PREFIX OGI: <http://ogi.eu/#>\n\<br>
PREFIX qb: <http://purl.org/linked-data/cube#>\n\<br>
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>\n\<br>
CONSTRUCT{\n\<br>
		?observation a qb:Observation;\n\<br>
				OGI:employee_number ?employee_number;\n\<br>
				OGI:municipality ?Municipality;\n\<br>
				OGI:time_period ?Time_period;\n\<br>
				OGI:economic_entities_in_operation_at_the_beginning_of_the_year_units ?Economic_entities_in_operation_at_the_beginning_of_the_year_units;\n\<br>
				OGI:counrty ?Country;\n\<br>
				OGI:county ?County;\n\<br>
				qb:dataSet OGI:EEOBY_ds;\n\<br>
				.\n\<br>
	}\n\<br>
WHERE {\n\<br>
BIND (uri(CONCAT('http://ogi.eu/#observations_',StrUUID())) AS ?observation).\n\<br>
}\n\<br>

</p>